core/src/main/java/com/linecorp/armeria/client/DefaultWebClient.java

```diff
@@ -44,36 +39,32 @@
 
     @Override
     public HttpResponse execute(HttpRequest req) {
-        URI uri;
-
-        if (isAbsoluteUri(req.path())) {
-            try {
-                uri = URI.create(req.path());
-            } catch (Exception ex) {
-                logger.warn("Failed to create URI: {}", req.path(), ex);
-                uri = null;
-            }
-        } else {
-            if (req.scheme() != null && req.authority() != null) {
+        if (Clients.isUndefinedUri(uri())) {
+            final URI uri;
+            if (isAbsoluteUri(req.path())) {
+                try {
+                    uri = URI.create(req.path());
+                } catch (Exception ex) {
+                    return failureResponse(req, new IllegalArgumentException(
+                            "Failed to create a URI: " + req.path(), ex));
+                }
+            } else if (req.scheme() != null && req.authority() != null) {
                 uri = req.uri();
             } else {
-                uri = null;
+                return failureResponse(req, new IllegalArgumentException("no authority: " + req.path()));
             }
-        }
-
-        if (uri != null) {
             final Endpoint endpoint = Endpoint.parse(uri.getAuthority());
             final String query = uri.getRawQuery();
             final String path = uri.getRawPath();
             final HttpRequest newReq = req.withHeaders(req.headers().toBuilder()
-                    .path(query == null ? path : path + '?' + query));
+                                                          .path(query == null ? path : path + '?' + query));
             return execute(endpoint, newReq);
         }
 
-        if (Clients.isUndefinedUri(uri())) {
-            final IllegalArgumentException cause = new IllegalArgumentException("no authority: " + req.path());
-            req.abort(cause);
-            return HttpResponse.ofFailure(cause);
+        if (isAbsoluteUri(req.path())) {
+            return failureResponse(req, new IllegalArgumentException(
+                    "Cannot send a request with an absolute path when the client is created with a base URI. " +
```

Comment:
`Cannot send a request with a ":path" header that contains a URI with authority, because the client was created with a base URI` ?

Reply:
Fixed. 😉